src/context/productContext.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useReducer } from "react";
+import { productReducer } from "../reducers/productReducer";
+
+const ProductContext = createContext();
```

Comment:
Here, you can give a fail-safe value in `createContext`. As Tanay has suggested that it is a good practice.

Reply:
* fall safe